Makefile

```diff
@@ -34,13 +35,23 @@ $(SAMPLEDATA):
 
 sampledata: $(SAMPLEDATA)
 
-bin/$(ZEEKPATH):
+bin:
 	@mkdir -p bin
+
+bin/$(ZEEKPATH): bin
 	@curl -L -o bin/$(ZEEKPATH).zip \
 		https://github.com/brimsec/zeek/releases/download/$(ZEEKTAG)/zeek-$(ZEEKTAG).$$(go env GOOS)-$$(go env GOARCH).zip
 	@unzip -q bin/$(ZEEKPATH).zip -d bin \
 		&& mv bin/zeek bin/$(ZEEKPATH)
 
+bin/mockgen-$(MOCKGEN_VERSION):
+	go get -d github.com/golang/mock/mockgen@$(MOCKGEN_VERSION)
+	go build -o bin/mockgen-$(MOCKGEN_VERSION) github.com/golang/mock/mockgen
+	ln -s mockgen-$(MOCKGEN_VERSION) bin/mockgen
+
+generate: bin/mockgen-$(MOCKGEN_VERSION)
+	@PATH=$(PATH):$(CURDIR)/bin go generate ./...
+	git diff --exit-code .
```

Comment:
Good idea.

Reply:
And since github.com/golang/mock is in go.mod, we don't need MOCKGEN_VERION or the bin/mockgen-$(MOCKGEN_VERSION) goal. We can just do this:
```make
generate:
	GOBIN=bin go install github.com/golang/mock/mockgen
	PATH=$(CURDIR)/bin:$(PATH) go generate ./...

test-generate: generate
	git diff --exit-code
```
Note that $(CURDIR)/bin is first in PATH; that's important.

Reply:
@nwt really slick